Beams/Logic/vtkSlicerIECTransformLogic.cxx

```diff
@@ -298,15 +333,145 @@ bool vtkSlicerIECTransformLogic::GetTransformBetween(CoordinateSystemIdentifier
+    std::copy( fromFramePath.begin(), fromFramePath.end(), fromFrameVector.begin());
+
+    outputTransform->Identity();
+    outputTransform->PostMultiply();
+    for ( size_t i = 0; i < fromFrameVector.size() - 1; ++i)
+    {
+      CoordinateSystemIdentifier parent, child;
+      child = fromFrameVector[i];
+      parent = fromFrameVector[i + 1];
+      vtkMRMLLinearTransformNode* fromTransform = this->GetTransformNodeBetween( child, parent);
+
+      vtkDebugMacro("GetTransformBetween: Checking affine transformation " 
+        << "\"" << this->CoordinateSystemsMap[child] << "\" -> " 
+        << "\"" << this->CoordinateSystemsMap[parent] << "\"");
+
+      if (fromTransform)
+      {
+        vtkNew<vtkMatrix4x4> mat;
+        fromTransform->GetMatrixTransformToParent(mat);
+        outputTransform->Concatenate(mat);
+
+        vtkDebugMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" is valid");
+      }
+      else
+      {
+        vtkErrorMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" is invalid");
+        return false;
+      }
+    }
+
+    for ( size_t i = 0; i < toFrameVector.size() - 1; ++i)
+    {
+      CoordinateSystemIdentifier parent, child;
+      parent = toFrameVector[i];
+      child = toFrameVector[i + 1];
+      vtkMRMLLinearTransformNode* toTransform = this->GetTransformNodeBetween( child, parent);
+
+      vtkDebugMacro("GetTransformBetween: Checking affine transformation " 
+        << "\"" << this->CoordinateSystemsMap[parent] << "\" -> " 
+        << "\"" << this->CoordinateSystemsMap[child] << "\"");
+
+      if (toTransform)
+      {
+        vtkNew<vtkMatrix4x4> mat;
+        toTransform->GetMatrixTransformFromParent(mat);
+        mat->Invert();
+        outputTransform->Concatenate(mat);
+
+        vtkDebugMacro("GetTransformBetween: Transform node to frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" from " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" is valid");
+      }
+      else
+      {
+        vtkErrorMacro("GetTransformBetween: Transform node from frame " 
+          << "\"" << this->CoordinateSystemsMap[child] << "\" to frame " 
+          << "\"" << this->CoordinateSystemsMap[parent] << "\" is invalid");
+        return false;
+      }
+    }
+
     return true;
   }
 
   vtkErrorMacro("GetTransformBetween: Failed to get transform " << this->GetTransformNodeNameBetween(fromFrame, toFrame));
   return false;
 }
+
+//-----------------------------------------------------------------------------
+bool vtkSlicerIECTransformLogic::GetPathToRoot( CoordinateSystemIdentifier frame, 
+  CoordinateSystemsList& path)
+{
+  bool found;

```

Comment:
always initialize variables (to ensure there is no difference between debug-mode and release-mode behavior) and move the variable to as close as possible where it is first used

Reply:
Fixed